bravado/mapping/param.py

```diff
@@ -128,6 +132,73 @@ def marshal_param(param, value, request):
             format(location))
 
 
+def unmarshal_param(param, request):
+    """Unmarshal the given parameter from the passed in request like object.
+
+    :type param: :class:`bravado.mapping.param.Param`
+    :type request: :class:`bravado.mapping.request.RequestLike`
+    """
+    param_spec = get_param_type_spec(param)
+    location = param.location
+    cast_param = partial(cast_request_param, param_spec['type'], param.name)
+
+    if location == 'path':
+        raw_value = cast_param(request.path.get(param.name, None))
+    elif location == 'query':
+        raw_value = cast_param(request.params.get(param.name, None))
+    elif location == 'header':
+        raw_value = cast_param(request.headers.get(param.name, None))
+    elif location == 'formData':
+        if param_spec['type'] == 'file':
+            raw_value = request.params.get(param.name, None)
+        else:
+            raw_value = cast_param(request.params.get(param.name, None))
+    elif location == 'body':
+        # TODO: verify content-type header
+        raw_value = request.json()
+    else:
+        raise SwaggerMappingError(
+            "Don't know how to unmarshal_param with location {0}".
+            format(location))
+
+    if param_spec['type'] == 'array' and location != 'body':
+        raw_value = unmarshal_collection_format(param_spec, raw_value)
+
+    validate_schema_object(param_spec, raw_value)
+    value = unmarshal_schema_object(param.swagger_spec, param_spec, raw_value)
+    return value
+
+
+CAST_TYPE_TO_FUNC = {
+    'integer': int,
+    'number': float,
+    'boolean': bool
+}
+
+
+def cast_request_param(param_type, param_name, param_value):
+    """Try to cast a request param (e.g. query arg, POST data) from a string to
+    its specified type in the schema. This allows validating non-string params.
+
+    :param param_type: name of the type to be casted to
+    :type  param_type: string
+    :param param_name: param name
+    :type  param_name: string
+    :param param_value: param value
+    :type  param_value: string
+    """
+    if param_value is None:
+        return None
+
+    try:
+        return CAST_TYPE_TO_FUNC.get(param_type, lambda x: x)(param_value)
```

Comment:
I don't think of any type which can be used just as `lambda x: x`. How about throwing a keyError instead `CAST_TYPE_TO_FUNC[param_type]`?